tests/test_schemas.py

```diff
@@ -797,27 +826,65 @@ def test_manually_routing_generic_view(self):
 
         generator = SchemaGenerator(title='Naming Colisions', patterns=patterns)
 
-        with pytest.raises(ValueError):
-            generator.get_schema()
+        schema = generator.get_schema()
+
+        self._verify_cbv_links(schema['test']['delete'], '/test/delete/')
+        self._verify_cbv_links(schema['test']['put'], '/test/put/')
+        self._verify_cbv_links(schema['test']['get'], '/test/get/')
+        self._verify_cbv_links(schema['test']['update'], '/test/update/')
+        self._verify_cbv_links(schema['test']['retrieve'], '/test/retrieve/')
+        self._verify_cbv_links(schema['test'], '/test', suffixes=(None, '0', None, '0'))
 
     def test_from_router(self):
         patterns = [
             url(r'from-router', include(naming_collisions_router.urls)),
         ]
 
         generator = SchemaGenerator(title='Naming Colisions', patterns=patterns)
+        schema = generator.get_schema()
+        desc = schema['detail_0'].description  # not important here
 
-        with pytest.raises(ValueError):
-            generator.get_schema()
+        expected = coreapi.Document(
+            url='',
+            title='Naming Colisions',
+            content={
+                'detail': {
+                    'detail_export': coreapi.Link(
+                        url='/from-routercollision/detail/export/',
+                        action='get',
+                        description=desc)
+                },
+                'detail_0': coreapi.Link(
+                    url='/from-routercollision/detail/',
+                    action='get',
+                    description=desc
+                )
+            }
+        )
 
+        assert schema == expected
 
-def test_is_list_view_recognises_retrieve_view_subclasses():
```

Comment:
We need this test case.

Reply:
strange its got removed, I will add it ofc